src/index.test.js

```diff
@@ -0,0 +1,137 @@
+  beforeEach(() => {
+    originalWriteFile = fsp.writeFile;
+    virtualFS = {};
+    fsp.writeFile = jest.fn().mockImplementation((file, contents) => {
+      virtualFS[file] = contents;
+      Promise.resolve();
+    });
+  });
+
+  afterEach(() => {
+    fsp.writeFile = originalWriteFile;
+  });
+
+  test('processFrontmatterFiles does the things it should do.', async () => {
+    await processFrontmatterFiles({
+      inputFilePatterns: ['test/*.md'],
+      dataFilePatterns: [
+        'test/*.yaml',
+        'test/*.json',
+        'test/*.yml',
+        'test/*.json5',
+      ],
+      data: { author: 'Lea Rosema' },
+      addCreated: false,
+      addModified: false,
+    });
+    expect(virtualFS['test/lea.md']).toBeDefined();
+    const md = MatterParser.fromString(virtualFS['test/lea.md']);
+    expect(md.metaData.author).toBe('Lea Rosema');
+    expect(md.metaData.gender).toBe('female');
+    expect(md.metaData.hobbies).toEqual(['sex', 'drugs', 'rocknroll']);
+    expect(md.content).toBe(
+      '\n# Hello World!\n\nLorem ipsum dolor sit amet.\n'
+    );
+  });
+
+  test('processFrontmatterFiles does the things it should do. Also add created and modified times.', async () => {
+    await processFrontmatterFiles({
+      inputFilePatterns: ['test/*.md'],
+      dataFilePatterns: [
+        'test/*.yaml',
+        'test/*.json',
+        'test/*.yml',
+        'test/*.json5',
+      ],
+      data: { author: 'terabaud' },
+      addCreated: true,
+      addModified: true,
+    });
+    expect(virtualFS['test/lea.md']).toBeDefined();
+    const md = MatterParser.fromString(virtualFS['test/lea.md']);
+    expect(md.metaData.author).toBe('terabaud');
+    expect(md.metaData.gender).toBe('female');
+    expect(md.metaData.hobbies).toEqual(['sex', 'drugs', 'rocknroll']);
+    expect(md.content).toBe(
+      '\n# Hello World!\n\nLorem ipsum dolor sit amet.\n'
+    );
+    expect(md.metaData.created).toBeDefined();
+    expect(md.metaData.modified).toBeDefined();
+  });
+
+  test('processFrontmatterFiles does the things it should do. Also add modified time.', async () => {
+    await processFrontmatterFiles({
+      inputFilePatterns: ['test/*.md'],
+      dataFilePatterns: [
+        'test/*.yaml',
+        'test/*.json',
+        'test/*.yml',
+        'test/*.json5',
+      ],
+      data: { author: 'Lea Rosema' },
+      addCreated: false,
+      addModified: true,
+    });
+    expect(virtualFS['test/lea.md']).toBeDefined();
+    const md = MatterParser.fromString(virtualFS['test/lea.md']);
+    expect(md.metaData.author).toBe('Lea Rosema');
+    expect(md.metaData.gender).toBe('female');
+    expect(md.metaData.hobbies).toEqual(['sex', 'drugs', 'rocknroll']);
+    expect(md.content).toBe(
+      '\n# Hello World!\n\nLorem ipsum dolor sit amet.\n'
+    );
+    expect(md.metaData.modified).toBeDefined();
+  });
+
+  test('processFrontmatterFiles does the things it should do. Also add created times.', async () => {
```

Comment:
I wouldn't test the same things in multiple tests. Tests are all about giving a signal for potential failure, and if all your tests have similar code, and they all fail, that's a noisy signal. So I'd break up these tests and make them (more) unique.

Reply:
Yeah, I can maybe remove the test about detailed meta data content (except modified/created) inside the test because this is already tested in the matter-parser test.